src/shims/mod.rs

```diff
@@ -27,9 +27,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         // There are some more lang items we want to hook that CTFE does not hook (yet).
         if this.tcx.lang_items().align_offset_fn() == Some(instance.def.def_id()) {
-            // FIXME: return a real value in case the target allocation has an
-            // alignment bigger than the one requested.
-            let n = u128::max_value();
+
+            let n = {
+                let ptr = this.force_ptr(this.read_scalar(args[0])?.not_undef()?)?;
```

Comment:
This will fail when passing in an integer. That's not what we want, is it?

Reply:
Not in general, but for now these are the common users

Reply:
Then there should be a `FIXME`. This is kind-of a regression though, the old code worked fine for integers, so I'd prefer if we could not regress.